goth/runner/download/__init__.py

```diff
@@ -82,107 +83,35 @@ def _create_cache_dir(self, asset_id: str) -> Path:
-
-            relation_to_url = self._parse_link_header(response.headers["Link"])
-            logger.debug("_search_with_pagination. relation_to_url=%s", relation_to_url)
-            next_url = relation_to_url.get("next")
-            if next_url:
-                logger.debug("_search_with_pagination. next_url=%s", next_url)
-                response = self.session.get(next_url)
-            else:
-                return None
-
 
 class ArtifactDownloader(GithubDownloader):
     """Downloader for GitHub Actions artifacts using GitHub's REST API."""
 
     def _get_workflow(self, workflow_name: str) -> dict:
         """Query the workflow on GitHub Actions."""
-        url = f"{self.repo_url}/actions/workflows"
-        logger.debug("Fetching workflows. url=%s", url)
-        response = self.session.get(url)
-        response.raise_for_status()
-
-        workflows = response.json()["workflows"]
-        logger.debug("workflows=%s", workflows)
-        workflow = next(filter(lambda w: w["name"] == workflow_name, workflows))
+        logger.debug("Fetching workflows. name=%s", workflow_name)
+        workflow = next(filter(lambda w: w["name"] == "CI", self.gh_api.actions.list_repo_workflows().workflows))
         logger.debug("workflow=%s", workflow)
+
         return workflow
 
     def _get_latest_run(
         self, workflow: dict, branch: str, commit: Optional[str] = None
     ) -> dict:
         """Filter out the latest workflow run."""
         workflow_id = workflow["id"]
-        url = f"{self.repo_url}/actions/workflows/{workflow_id}/runs"
-        params = {"status": "completed"}
-        if not commit:
-            params["branch"] = branch
+        logger.debug("Fetching workflow runs. workflow_id=%s", workflow_id)
 
-        request = self.session.prepare_request(
-            requests.Request("GET", url, params=params)
-        )
-        logger.debug("Fetching workflow runs. url=%s", request.url)
-
-        def _filter_workflows(response: requests.Response) -> Optional[dict]:
-            workflow_runs = response.json()["workflow_runs"]
-            if commit:
-                return next(
-                    filter(lambda r: r["head_sha"].startswith(commit), workflow_runs),
-                    None,
-                )
-            else:
-                return workflow_runs[0]
-
-        workflow_run = self._search_with_pagination(request, _filter_workflows)
-        logger.debug("workflow_run=%s", workflow_run)
-        return workflow_run
+        if commit:
+            paged_workflow_runs = paged(self.gh_api.list_workflow_runs, workflow_id, status="completed")
+
+            for page in paged_workflow_runs:
+                workflow_runs = next(filter(lambda r: r["head_sha"].startswith(commit), page.workflow_runs), None)
+
+                if workflow_runs:
+                    return workflow_runs
+
+        return self.gh_api.list_workflow_runs(workflow_id, branch=branch, status="completed").workflow_runs[0]
```

Comment:
When using the standalone downloader script from this repo (`./scripts/download_artifacts.py)` from the repo root) this throws an exception for me:
```
Traceback (most recent call last):
  File "./scripts/download_artifacts.py", line 64, in <module>
    downloader.download(
  File "/home/kuba/code/python/goth-ghapi/goth/runner/download/__init__.py", line 193, in download
    latest_run = self._get_latest_run(workflow, branch, commit)
  File "/home/kuba/code/python/goth-ghapi/goth/runner/download/__init__.py", line 126, in _get_latest_run
    return self.gh_api.list_workflow_runs(
  File "/home/kuba/.virtualenvs/ghapi/lib/python3.8/site-packages/ghapi/core.py", line 116, in __getattr__
    def __getattr__(self,k): return self.groups[k] if 'groups' in vars(self) and k in self.groups else stop(AttributeError(k))
  File "/home/kuba/.virtualenvs/ghapi/lib/python3.8/site-packages/fastcore/basics.py", line 202, in stop
    raise e
AttributeError: list_workflow_runs
```
I'll try to find a moment next week to see why this is happening, but I'm mentioning it here if you happen to get there first. :)

Reply:
found an issue with [ghapi.actions.download_artifact](https://ghapi.fast.ai/fullapi.html), it's trying to decode the download archive which is breaking. For now, revert back to `request.session`  for download.

ref: https://github.com/fastai/ghapi/issues/22